### PR TITLE
fixes #170 - incompatible (can't compile) with IDEA-141.157.3

### LINKED ISSUE #170: incompatible (can't compile) with IDEA-141.157.3
error in HaxeStructureViewFactory because:
TreeBasedStructureViewBuilder.createStructureViewModel() is removed in [this commit](https://github.com/JetBrains/intellij-community/commit/f1fa0838a03e3ae3b5f16d74ee99d017f90c8483)

error in HaxeRegularExpressionImpl because:
HaxeRegularExpressionImpl.supportsExtendedHexCharacter  & supportsNamedGroupRefSyntax is added in [this commit](https://github.com/JetBrains/intellij-community/commit/478790592021e30ccf37056175a87b88e173b735) & [this commit](https://github.com/JetBrains/intellij-community/commit/733be82403903d8616366627e3e841ceb84c3c61)

### PR DESCRIPTION
**Pull request by giabao**

### CODE REVIEW

**Comment by as3boyan** [comment]
@EBatTiVo First commit is definitely required for IDEA 14.1. Second one seems fine too, not sure why to add "build" folder to .gitignore, maybe some IDEA 14.1 specific stuff.

File: src/com/intellij/plugins/haxe/ide/structure/HaxeStructureViewElement.java [hunk 1]
```diff
@@ -67,12 +68,14 @@ public boolean canNavigateToSource() {
     return myElement instanceof NavigationItem && ((NavigationItem)myElement).canNavigateToSource();
   }
 
+  //FIXME override @NotNull with a @Nullable!
```
Comment (as3boyan) [inline src/com/intellij/plugins/haxe/ide/structure/HaxeStructureViewElement.java]: so interface method is marked as NotNull?

**Comment by EBatTiVo** [comment]
I have no problem with the functionality changes.  I have held off because we are in the middle of getting a single source base to compile for all flavors.  It's almost done, so that will make it easier to compile, and maybe I won't have to backport this change.  I'll probably hand-merge this change to the master branch once I've confirmed the multi-build strategy is working.

**Comment by as3boyan** [comment]
@EBatTiVo Great, so we will have build server, where everybody can download nightly build for both versions of IDEA?

**Comment by fzzr-** [comment]
Nightly builds will be cool! :+1:

**Comment by EBatTiVo** [comment]
> so we will have build server, where everybody can download nightly build for both versions of IDEA?

Yes and no.  On the TiVo/intellij-haxe github repository: Using Travis,ci, we will have continuous integration builds for versions 13 and 14 (and, most likely, 14.1) that will build and run all of the unit tests against every pull request (before it's pulled) and after every merge or push to the master branch.  Release builds are not automatically made from that system.  

On top of that, we will run a cron job on a computer locally (at TiVo) to automatically build and post releases at (I believe) weekly intervals.  We haven't looked into any automated pull requests to the JetBrains github repo or setting any of this up there.  Nor have we looked into automating releases to the IDEA plugin repository.

**Comment by fzzr-** [comment]
Maybe is possible to configure Travis-CI for upload build to you server. I don't know is it possible and where store auth-data-for-your-server but this is idea.

**Comment by EBatTiVo** [comment]
@fzzr- We probably *could* set up automatic uploads, since we decide what the script actually does.  The trouble with that is that Travis builds every pull request and every push on every branch, and we don't want to keep all of those (unmerged) builds.  We've got a different solution for releases.

**Comment by as3boyan** [comment]
Can travis check for example if commit message contains "#Build" or something like that if that is not too complicated

**Comment by as3boyan** [comment]
It doesn't seem to be provided natively, but 
Some guy found a way to do this by getting last commit message from `git log`
https://github.com/travis-ci/travis-ci/issues/965

What do you think about it?

**Comment by as3boyan** [comment]
Or even shorter `git log -n 1 --oneline`
which outputs only this:
`0cb9056 Merge branch 'master' into masterIdea14`

**Comment by as3boyan** [comment]
Upload to JetBrains plugin repository script(Unix)
https://github.com/MoreUnit/org.moreunit.intellij.plugin/blob/master/upload-plugin.sh

**Comment by yanhick** [comment]
Hi,
I'm working on getting the nighly builds to work. I've been testing it for the last week on one of my repo before making it "official" with a Tivo nightly build repo:
https://github.com/yanhick/intellij-haxe-nightly-builds/releases

It leverages Travis "deploy" feature to generate the intellij-haxe.jar every day using a little script that you can find here:
https://github.com/yanhick/intellij-nightly

What's left to be done:
- build multiple intellij-haxe.jar, one for each supported version. This is dependant on this pull request : https://github.com/TiVo/intellij-haxe/pull/195
- Figure out the right format for the timestamp of the release, they appear out of order for now

Once everything is set-up, I'll document the nightly builds in the readme

**Comment by yanhick** [comment]
@as3boyan, 
interesting script, could be nice to automate release on  JetBrains repo too. Do you know if they have multiple release channels ? For example would it be possible to automate building "alpha" or "beta" releases ?

**Comment by as3boyan** [comment]
@yanhick I don't think so. Haven't seen such options.

**Comment by as3boyan** [comment]
I wrote a script to upload jar to JetBrains repo, works fine

**Comment by EBatTiVo** [comment]
Superceded by #236.

**Pull request closed without merging**